client/my-sites/checkout/composite-checkout/payment-method-processors.js

```diff
@@ -256,12 +256,9 @@ export async function payPalProcessor(
 ) {
 	const { createUserAndSiteBeforeTransaction } = transactionOptions;
 	const { protocol, hostname, port, pathname } = parseUrl( window.location.href, true );
-	const successUrl = formatUrl( {
-		protocol,
-		hostname,
-		port,
-		pathname: getThankYouUrl(),
-	} );
+
+	const successUrl = resolveUrl( window.location.href, getThankYouUrl() );
```

Comment:
@sirbrillig this small yet significant change. Would really appreciate your approval.

Reply:
Thanks for calling that out. It looks fine to me! I tested a PayPal purchase and the successUrl looks correct.